### PR TITLE
BX-4153 Place Order

### PR DESCRIPTION
**Pull request by aspin**

### CODE REVIEW

File: bxserum/provider/common.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+func DefaultRPCOpts(endpoint string) (RPCOpts, error) {
+	privateKey, err := transaction.LoadPrivateKeyFromEnv()
+	if err != nil {
+		return RPCOpts{}, err
+	}
+	return RPCOpts{
+		Endpoint:   endpoint,
+		Timeout:    time.Second * 7,
```
Comment (rafaelvanoni) [inline bxserum/provider/common.go]: Nit: use a constant for `7`

File: bxserum/provider/grpc.go [hunk 2]
```diff
@@ -3,46 +3,52 @@ package provider
...
-	requestID utils.RequestID
+	apiClient  pb.ApiClient
+	privateKey solana.PrivateKey
 }
 
-// Connects to Mainnet Serum API
+// NewGRPCClient connects to Mainnet Serum API
 func NewGRPCClient() (*GRPCClient, error) {
-	return NewGRPCClientWithEndpoint("174.129.154.164:1811")
+	opts, err := DefaultRPCOpts("174.129.154.164:1811")
```
Comment (rafaelvanoni) [inline bxserum/provider/grpc.go]: Nit: use constant for ip:port

File: bxserum/provider/grpc.go [hunk 3]
```diff
@@ -3,46 +3,52 @@ package provider
...
+// NewGRPCClientWithOpts connects to custom Serum API
+func NewGRPCClientWithOpts(opts RPCOpts) (*GRPCClient, error) {
+	conn, err := grpc.Dial(opts.Endpoint, grpc.WithTransportCredentials(insecure.NewCredentials()))
 	if err != nil {
 		return nil, err
 	}
 	return &GRPCClient{
-		apiClient: pb.NewApiClient(conn),
-		requestID: utils.NewRequestID(),
+		apiClient:  pb.NewApiClient(conn),
```
Comment (rafaelvanoni) [inline bxserum/provider/grpc.go:41]: gofmt ? extra space before `pb`

File: bxserum/provider/grpc.go [hunk 4]
```diff
@@ -66,16 +73,56 @@ func (g *GRPCClient) GetTradesStream(ctx context.Context, market string, limit u
...
 }
+
+// PostOrder returns a partially signed transaction for placing a Serum market order. Typically, you want to use SubmitOrder instead of this.
+func (g *GRPCClient) PostOrder(ctx context.Context, owner, payer, market string, side pb.Side, amount, price float64, opts PostOrderOpts) (*pb.PostOrderResponse, error) {
+	return g.apiClient.PostOrder(ctx, &pb.PostOrderRequest{
+		OwnerAddress:      owner,
+		PayerAddress:      payer,
+		Market:            market,
+		Side:              side,
+		Type:              []pb.OrderType{pb.OrderType_OT_LIMIT},
```
Comment (rafaelvanoni) [inline bxserum/provider/grpc.go]: Why always a limit order?

**Review by rafaelvanoni — APPROVED** [review]
lgtm

File: bxserum/provider/grpc.go (hunk 4 above)
Comment (aspin) [reply bxserum/provider/grpc.go]: it's the only one we support right now anyway. I suppose I will allow the SDK to specify any type, though non limit types will be rejected by the API

File: bxserum/provider/grpc.go (hunk 3 above)
Comment (aspin) [reply bxserum/provider/grpc.go:41]: gofmt doesn't change anything. On further inspection it's to align with the field below

**Pull request merged**